app/src/androidTest/java/com/google/samples/apps/topeka/activity/SignInActivityTest.java

```diff
@@ -65,15 +71,33 @@ public void checkFab_initiallyNotDisplayed() {
     }
 
     @Test
+    public void signIn_checkIfEditTextViewsAreInitiallyEmpty() {
+        onView(withId(R.id.first_name)).check(matches(withText(isEmptyOrNullString())));
+        onView(withId(R.id.last_initial)).check(matches(withText(isEmptyOrNullString())));
+    }
+
+    @Test
     public void signIn_performSuccessful() {
         inputData();
-        onView(withId(R.id.done)).check(matches(isDisplayed()));
+        onView(withId(R.id.done)).check(matches(isDisplayed())).perform(click());
+        onView(withId(R.id.toolbar_player)).check(matches(isDisplayed()));
+    }
+
+    @Test
+    public void signIn_chooseAvatar() {
+        for (int i = 0; i < 16; i++) {
+            onData(anything())
+                    .inAdapterView(withId(R.id.avatars))
+                    .atPosition(15)
+                    .check(matches(isDisplayed()))
+                    .perform(click()).check(matches(isChecked()));
+        }
     }
 
     @Test
     public void signIn_withLongLastName() {
         inputData();
-        onView(withId(R.id.last_initial)).perform(typeText("somelongtext"), closeSoftKeyboard());
+        onView(withId(R.id.last_initial)).perform(typeText(valueOf(R.string.first_name)), closeSoftKeyboard());
```

Comment:
nothing more than you. Just don't like to use in tests. layouts, classes hard typed string like "somelongstring". Instead of this, I used an existing one.